client/API.md

```diff
@@ -0,0 +1,536 @@
+| start     | Earliest date for which data will be recorded. RFC3339Nano formatted.                                            |
+| stop      | Latest date for which data will be recorded. If not specified uses the current time. RFC3339Nano formatted data. |
+| past      | Duration string indicating how far into the past to start recording data relative to the stop date.              |
+| cluster   | Name of a configured InfluxDB cluster. If empty uses the default cluster.                                    |
+
+Only one of `start` or `past` may be specified.
+
+##### Query
+
+| Parameter | Purpose                                                                       |
+| --------- | -------                                                                       |
+| type      | Type of recording, `stream` or `batch`.                                   |
+| query     | Query to execute.                                                         |
+| cluster   | Name of a configured InfluxDB cluster. If empty uses the default cluster. |
+
+>NOTE: A recording itself is typed as either a stream or batch recording and can only be replayed to a task of a corresponding type.
+Therefore when you record the result of a raw query you must specify the type recording you wish to create.
+
+
+#### Example
+
+Create a recording using the `stream` method
+
+```
+POST /recording
+{
+    "method" : "stream",
+    "name" : "TASK_NAME",
+    "duration" : "10m"
+}
+```
+
+
+Create a recording using the `batch` method specifying a start time.
+
+```
+POST /recording
+{
+    "method" : "batch",
+    "name" : "TASK_NAME",
+    "start" : "2006-01-02T15:04:05Z07:00"
+}
+```
+
+Create a recording using the `batch` method specifying a past duration.
+
+```
+POST /recording
+{
+    "method" : "batch",
+    "name" : "TASK_NAME",
+    "past" : "7d"
+}
+```
+
+Create a recording using the `query` method specifying a `stream` type.
+
+```
+POST /recording
+{
+    "method" : "query",
+    "query" : "SELECT mean(usage_idle) FROM cpu WHERE time > now() - 1h GROUP BY time(10m)",
+    "type" : "stream"
+}
+```
+
+Create a recording using the `query` method specifying a `batch` type.
+
+```
+POST /recording
+{
+    "method" : "query",
+    "query" : "SELECT mean(usage_idle) FROM cpu WHERE time > now() - 1h GROUP BY time(10m)",
+    "type" : "batch"
+}
+```
+
+
+#### Response
+
+All recordings are assigned an ID which is returned in this format.
+
+```
+{
+    "id" : "e24db07d-1646-4bb3-a445-828f5049bea0"
+}
+```
+
+| Code | Meaning                             |
+| ---- | -------                             |
+| 200  | Success, the recording has started. |
+
+### Wait for Recording
+
+In order to determine when a recording has finished you must request it by making a GET request to `/recording/RECORDING_ID`.
+A GET request for a recording will block until the recording is complete.
```

Comment:
I totally agree with @goller, and I think it would be nice to add recording progress status, something like:
```
> GET /recording/e24db07d-1646-4bb3-a445-828f5049bea0

{
   "status": "running", // will be "running" or "done"
   "pct": 0.8 // how much recorded , for example the recording duration is 100m and it's done only 80m
}
```